flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -100,6 +100,12 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
             }
             this.tryShowMessage();
         };
+        this.onClrHeld = () => {
```

Comment:
Please follow the same logic as the onClr event...

Also when holding down the clear button, does it clear the first character and then the entire scratchpad?

How does it behave when an type I or type II message is displayed?

Reply:
> EDIT: The follow needs to be true
> 
> When holding clear button, our onClr event should not be triggered.

That _cannot_ be true, because:
- we cannot tell in advance whether the user will hold the button down for longer or not,
- the real MCDU does not do this for the same reason

The real MCDU clears the last character on the rising edge, as we do by triggering onClr, and then the rest later if it is held down. Additionally, if the scratchpad is empty, the rising trigger (onClr) adds a CLR to the pad, and holding down has no further affect in this PR. I don't have a video reference for this behaviour, so can't confirm it 100%, but it seems to match the FCOM.

> Also now logic for type I and type II messages need to be added to this function.

This part I now see is a problem. The user input buffer is shared with the FMC messages, which is a design mistake we should probably rectify to simplify the logic, or we can hax around it with additional logic in onClrHeld.

I am busy with work on the new ND and FMGC thus I don't have more time to work on the scratchpad logic for now, so I'll close this.

Reply:
True,

when pressing the button the event onClr fires and when then holding the button, the event onClrHold gets executed next.

IMO it still needs to account for a message being displayed when there is only one char on the scratchpad beforehand.

Reply:
Yes, you are correct and I don't think it does that correctly at the moment. I might get back to this one day, or somebody else might.

Reply:
Maybe I'll pick up on this one

Reply:
> Maybe I'll pick up on this one

Yep, if you can that'd be good. 👍